File: Algorithms/557. Reverse Words in a String III/swift/solution.reverse.swift
class Solution {
    func reverseWords(_ s: String) -> String {
        var result: [String] = []
        var word: String = ""
        for ch in s {
            if ch == " " {
                result.append(word)
                word = ""
            } else {
                word = String(ch) + word
            }
        }
        result.append(word)
        return result.joined(separator: " ")
    }
}

File: Algorithms/557. Reverse Words in a String III/python/solution.reverse.py
class Solution:
    def reverseWords(self, s: str) -> str:
        result: list[str] = []
        word: str = ""
        for ch in s:
            if ch == " ":
                result.append(word[::-1])
                word = ""
            else:
                word += ch
        result.append(word[::-1])
        return " ".join(result)
